src/iexamine.cpp

```diff
@@ -70,11 +70,10 @@ void iexamine::gaspump(player &p, const tripoint &examp)
             ///\EFFECT_DEX decreases chance of spilling gas from a pump
             if( one_in(10 + p.get_dex()) ) {
                 add_msg(m_bad, _("You accidentally spill the %s."), item_it->type_name().c_str());
-                item spill( item_it->typeId(), calendar::turn );
-                const int spill_min = item_it->liquid_charges( 1 );
                 ///\EFFECT_DEX decreases amount of gas spilled from a pump
-                const int spill_max = item_it->liquid_charges( 1 ) * 8.0 / std::max( 1, p.get_dex() );
-                spill.charges = rng( spill_min, spill_max );
+                const int qty = rng( item_it->liquid_charges( 1 ),
+                                     item_it->liquid_charges( 1 ) * 8.0 / std::max( 1, p.get_dex() ) );
+                item spill( item_it->type, calendar::turn, qty );
```

Comment:
Why not `item::split( int qty )`? It prevents both exposing `item::type` on explicit construction and reassignment of non-encapsulated `item::charges` (here and below).